packages/react/src/components/Pagination/usePagination.ts

```diff
@@ -1,103 +1,74 @@
-import { useEffect, useState } from 'react';
+import { useMemo } from 'react';
+import type { Dispatch, MouseEvent, SetStateAction } from 'react';
+import type { ButtonProps } from '../Button';
 
-import type { PaginationProps } from './Pagination';
+const getSteps = (now: number, max: number, show: number) => {
+  const offset = (show - 1) / 2;
+  const start = Math.min(Math.max(now - Math.floor(offset), 1), max - show + 1);
+  const end = Math.min(Math.max(now + Math.ceil(offset), show), max);
+  const pages = Array.from({ length: end + 1 - start }, (_, i) => i + start);
 
-type GetStepsProps = Pick<
-  PaginationProps,
-  'compact' | 'currentPage' | 'totalPages'
->;
+  if (show > 4 && start > 1) pages.splice(0, 2, 1, 0);
+  if (show > 3 && end < max) pages.splice(-2, 2, 0, max);
 
-const getSteps = ({
-  compact,
-  currentPage,
-  totalPages,
-}: GetStepsProps): ('ellipsis' | number)[] => {
-  /**  Number of always visible pages at the start and end. */
-  const boundaryCount = 1;
-
-  /** Number of always visible pages before and after the current page. */
-  const siblingCount = compact ? 0 : 1;
-
-  const range = (start: number, end: number) =>
-    Array.from({ length: end - start + 1 }, (_, i) => start + i);
-
-  if (totalPages <= (boundaryCount + siblingCount) * 2 + 3)
-    return range(1, totalPages);
-
-  const startPages = range(1, boundaryCount);
-  const endPages = range(totalPages - boundaryCount + 1, totalPages);
-
-  const siblingsStart = Math.max(
-    Math.min(
-      currentPage - siblingCount,
-      totalPages - boundaryCount - siblingCount * 2 - 1,
-    ),
-    boundaryCount + 2,
-  );
-  const siblingsEnd = siblingsStart + siblingCount * 2;
-
-  return [
-    ...startPages,
-    siblingsStart - (startPages[startPages.length - 1] ?? 0) === 2
-      ? siblingsStart - 1
-      : 'ellipsis',
-    ...range(siblingsStart, siblingsEnd),
-    (endPages[0] ?? totalPages + 1) - siblingsEnd === 2
-      ? siblingsEnd + 1
-      : 'ellipsis',
-    ...endPages,
-  ];
+  return pages;
 };
 
-export type UsePaginationProps = Pick<
-  PaginationProps,
-  'compact' | 'totalPages'
-> &
-  Partial<Pick<PaginationProps, 'currentPage'>>;
+export type UsePaginationProps = {
+  currentPage: number;
+  setCurrentPage?: Dispatch<SetStateAction<number>>;
+  onChange?: (event: MouseEvent<HTMLElement>, page: number) => void;
+  totalPages: number;
+  showPages?: number;
+};
```

Comment:
Ref to other comments, suggest adding `size` here

Reply:
Have discussed this with Micke, and will keep `size` on `Pagination` for now, as this matches the pattern of all other components, and is less error-prone than sniffing child components.